### PR TITLE
fix(CORE/Script): Quest "A Pawn on the Eternal Board" can't finish and hung up the worldserver.exe

### PR DESCRIPTION
**Pull request by eyeofstorm**
## Changes Proposed:
-  fix the script "npc_anachronos_the_ancient" which will hung up the server.
-  fix some logic errors in script "npc_anachronos_the_ancient".

## Issues Addressed:

- Closes 

## Tests Performed:

- build without errors 
- no server hung up/crash
- quest "A Pawn on the Eternal Board" can finished.

## How to Test the Changes:

1. accept the quest "A Pawn on the Eternal Board". (quest id: 8519)
2. view the quest animation.
3. confirm there is no server hung up/crash and the quest can finished.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Kalimdor/zone_silithus.cpp [hunk 1]
```diff
@@ -535,18 +535,23 @@ class npc_anachronos_the_ancient : public CreatureScript
...
-                                    mob = player->FindNearestCreature(15423, 50, me);
+                                    if (mob->IsInWorld())
+                                    {
+                                        Map* map = mob->GetMap();
+                                        map->RemoveFromMap(mob, false);
+                                    }
+
+                                    mob = Fandral->FindNearestCreature(entries[i], 50.0f);
                                 }
                             }
```
Comment (pklloveyou) [inline src/server/scripts/Kalimdor/zone_silithus.cpp:549]:
```
                    {
                        std::list<Creature*> constructList;
                        me->GetCreatureListWithEntryInGrid(constructList, 15423, 100.0f);
                        me->GetCreatureListWithEntryInGrid(constructList, 15424, 100.0f);
                        me->GetCreatureListWithEntryInGrid(constructList, 15414, 100.0f);
                        me->GetCreatureListWithEntryInGrid(constructList, 15422, 100.0f);
                        if (!constructList.empty())
                            for (std::list<Creature*>::const_iterator itr = constructList.begin(); itr != constructList.end(); ++itr)
                                (*itr)->RemoveFromWorld();
                    }
```
Would it be better to write like this?

File: src/server/scripts/Kalimdor/zone_silithus.cpp [hunk 2]
```diff
@@ -532,21 +532,26 @@ class npc_anachronos_the_ancient : public CreatureScript
...
                     case 51:
                         {
-                            uint32 entries[4] = { 15423, 15424, 15414, 15422 };
-                            for (uint8 i = 0; i < 4; ++i)
+                            std::list<Creature*> constructList;
+
+                            me->GetCreatureListWithEntryInGrid(constructList, 15423, 100.0f);
+                            me->GetCreatureListWithEntryInGrid(constructList, 15424, 100.0f);
+                            me->GetCreatureListWithEntryInGrid(constructList, 15414, 100.0f);
+                            me->GetCreatureListWithEntryInGrid(constructList, 15422, 100.0f);
```
Comment (UltraNix) [inline src/server/scripts/Kalimdor/zone_silithus.cpp:540]: Every call of `GetCreatureListWithEntryInGrid` overrides the previous container, so only mobs from the last call are removed from world.

**Review by UltraNix — APPROVED** [review]

**Comment by mpfans** [comment]
testing

**Comment by mpfans** [comment]
tested, work now.good job!

**Comment by mpfans** [comment]
Please merge it first  @Nefertumm

**Comment by Kitzunu** [comment]
Thanks for the fix

**Pull request merged**